### PR TITLE
Google Calendar

### PR DESCRIPTION
**Pull request by michaeljevans**
-Added events controller
-Added Google Calendar service

### CODE REVIEW

File: app/controllers/view_parties_controller.rb [hunk 1]
```diff
@@ -4,9 +4,14 @@ def new
...
   def create
-    party = ViewParty.create(title: params[:title], duration: params[:duration], date: params[:date], time: params[:time], poster: params[:poster], user_id: current_user.id)
-    PartyGuest.create_invites(party.id, params[:viewing_party][:friend_ids])
-
+    party = current_user.view_parties.create(view_party_params)
+    PartyGuest.create_invites(party.id, params[:view_party][:friend_ids])
     redirect_to dashboard_path
   end
+
+  private
```
Comment (janegreene) [inline app/controllers/view_parties_controller.rb:12]: This is much more conventional, glad you could fix this.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-  has_many :party_guests
+  has_many :view_parties, dependent: :destroy
+  has_many :party_guests, dependent: :destroy
 
   def self.from_omniauth(params)
     user = find_or_create_by(uid: params[:uid])
-    user.name = params[:info][:name]
     user.email = params[:info][:email]
     user.access_token = params[:credentials][:token]
+    user.refresh_token = params[:credentials][:refresh_token]
```
Comment (janegreene) [inline app/models/user.rb:14]: Google API needs this for long lived visitors.

File: app/controllers/view_parties_controller.rb [hunk 3]
```diff
@@ -4,9 +4,14 @@ def new
   end
 
   def create
-    party = ViewParty.create(title: params[:title], duration: params[:duration], date: params[:date], time: params[:time], poster: params[:poster], user_id: current_user.id)
-    PartyGuest.create_invites(party.id, params[:viewing_party][:friend_ids])
-
+    party = current_user.view_parties.create(view_party_params)
+    PartyGuest.create_invites(party.id, params[:view_party][:friend_ids])
```
Comment (michaeljevans) [inline app/controllers/view_parties_controller.rb:8]: nice refactor with strong params

File: app/services/google_calendar_service.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    service.authorization.refresh!
+
+    event = Google::Apis::CalendarV3::Event.new(
+      summary: "Viewing party for #{view_party.title}",
+      description: 'BYOB!',
+      start: Google::Apis::CalendarV3::EventDateTime.new(date: view_party.date),
+      end: Google::Apis::CalendarV3::EventDateTime.new(date: view_party.date)
+    )
+
+    service.insert_event('primary', event)
```
Comment (janegreene) [inline app/services/google_calendar_service.rb:17]: Select users primary calendar

File: app/views/dashboard/show.html.erb [hunk 5]
```diff
@@ -1,43 +1,58 @@
...
+    <p>You currently have no friends.</p>
   <% else %>
     <%current_user.friends.each do |friend|%>
-      <%= friend.name %>
+      <%= friend.email %>
+      <br>
     <% end %>
   <% end %>
+  <br>
+  <%= form_tag '/friendships', method: :post do %>
```
Comment (janegreene) [inline app/views/dashboard/show.html.erb:18]: Form-tags need a do and end...the more you know!

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -11,7 +11,7 @@
   <body>
     <main>
       <% if current_user %>
-        <p style='font-size:20px;'>Welcome, <%= current_user.name %>!</p>
+        <p style='font-size:20px;'>Welcome, <%= current_user.email %>!</p>
```
Comment (janegreene) [inline app/views/layouts/application.html.erb:14]: Once we changed scope for calendar API, name was no longer returned to us.

**Pull request merged**